pkg/k8splan/watcher.go

```diff
@@ -365,7 +370,33 @@ func (w *watcher) updateSecret(core corecontrollers.Interface, secret *v1.Secret
 }
 
 func validateKC(ctx context.Context, config *rest.Config) error {
+	var (
+		conn *tls.Conn
+		err  error
+	)
+
 	config = rest.CopyConfig(config)
+	transportConfig, err := config.TransportConfig()
+	if err != nil {
+		return err
+	}
+	tlsConfig, err := transport.TLSConfigFor(transportConfig)
+	if err != nil {
+		return err
+	}
+
+	config.Transport = utilnet.SetTransportDefaults(&http.Transport{
+		DialTLS: func(network, addr string) (net.Conn, error) {
+			conn, err = tls.Dial(network, addr, tlsConfig)
+			return conn, err
+		},
+	})
+	config.WrapTransport = transportConfig.WrapTransport
+	config.Dial = transportConfig.Dial
+	// Overwrite TLS-related fields from config to avoid collision with
```

Comment:
Can you add some more context regarding why we had to do this (i.e. specifically call out how go behaves on Windows vs. Linux regarding transports)?